src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
The idea was to move Auth to be a service, which has more or less been accomplished.  We wanted to move away from invoking all handlers for all requests by default (for perf and to work better with multi-tenancy), while not blocking the ability for a scheme to inspect all requests.

If replacing the default `IAuthenticationSchemeProvider` is not an option, then you are only left with adding your own call to `IAuthenticationSchemeProvider.GetSchemeAsync("ASOS").ProcessRequestAsync()` somewhere, probably via middleware

If this is ends up being a common scenario, we can always add a new options flag like InvokeAllSchemesForAllRequests (aka SlowMoMode = true) for the single Auth middleware to call them all like before.